fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java

```diff
@@ -491,9 +384,12 @@ public Boolean processPayload(ResourcePayload t) {
                     logger.fine(logPrefix() + " Processing payload for '" + this.resourceType.getSimpleName() + "/" + t.getLogicalId() + "'");
                 }
 
-                // Check if this data would cause us to exceed the max object size. If
+                // Check if this data would cause us to exceed max object size. If
                 // so, close the current upload and start a new one
-                if (this.uploadId != null && (currentObjectSize + this.outputStream.size() > maxObjectSize
+                //
+                // @implNote outputStream.size() is not set for the current resource yet set at this point of the current Resource, comparing it
+                //  can lead to unintended results with double counting.
+                if (this.uploadId != null && (currentObjectSize > maxObjectSize
```

Comment:
Here is the point of discussion

Reply:
In prior it was comparing against an already populated outputStream.size()
note the outputstream is not accumulated until line #405